### PR TITLE
[arrow] Fix android

### PR DESCRIPTION
**Pull request by dg0yt**
Needed with NDK r26. Cherry-picked from #35851.

### CODE REVIEW

File: ports/arrow/portfile.cmake [hunk 1]
```diff
@@ -8,10 +8,10 @@ vcpkg_extract_source_archive(
     SOURCE_PATH
     ARCHIVE ${ARCHIVE_PATH}
     PATCHES
+        android.patch
         msvc-static-name.patch
         utf8proc.patch
         thrift.patch
-        fix-ci-error.patch
```
Comment (LilyWangLL) [inline ports/arrow/portfile.cmake:14]: Is this patch no longer needed? If it isn't, please also remove file `fix-ci-error.patch`.

**Review by LilyWangLL — CHANGES_REQUESTED** [review]

File: ports/arrow/portfile.cmake (hunk 1 above)
Comment (dg0yt) [reply ports/arrow/portfile.cmake:14]: The change is part of android.patch now. Patch removed.

**Review by LilyWangLL — APPROVED** [review]

File: ports/arrow/android.patch [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+ endif()
+diff --git a/cpp/src/arrow/vendored/musl/strptime.c b/cpp/src/arrow/vendored/musl/strptime.c
+index 41912fd..0ea36e9 100644
+--- a/cpp/src/arrow/vendored/musl/strptime.c
++++ b/cpp/src/arrow/vendored/musl/strptime.c
+@@ -18,7 +18,9 @@
+ #undef HAVE_LANGINFO
+ 
+ #ifndef _WIN32
++# if !(defined(__ANDROID__) && __ANDROID_API__ < 26)
```
Comment (BillyONeal) [inline ports/arrow/android.patch:22]: !(defined(__ANDROID__) && __ANDROID_API__ < 26)

is

!defined(__ANDROID__) || __ANDROID_API__ >= 26

LGTM

**Comment by BillyONeal** [comment]
Thanks :)

**Pull request merged**